src/components/new-safe/create/steps/StatusStep/useSafeCreationEffects.ts

```diff
@@ -19,36 +19,32 @@ const useSafeCreationEffects = ({
   const chainId = useChainId()
   const [pendingSafe, setPendingSafe] = usePendingSafe()
 
+  // Asynchronously wait for Safe creation
   useEffect(() => {
-    if (status === SafeCreationStatus.SUCCESS) {
-      trackEvent(CREATE_SAFE_EVENTS.CREATED_SAFE)
-
-      // Add the Safe and add names to the address book
-      if (pendingSafe && pendingSafe.safeAddress) {
-        dispatch(
-          updateAddressBook(
-            chainId,
-            pendingSafe.safeAddress,
-            pendingSafe.name,
-            pendingSafe.owners,
-            pendingSafe.threshold,
-          ),
-        )
-      }
-
-      // Asynchronously wait for Safe creation
-      if (pendingSafe?.safeAddress) {
-        pollSafeInfo(chainId, pendingSafe.safeAddress)
-          .then(() => setStatus(SafeCreationStatus.INDEXED))
-          .catch(() => setStatus(SafeCreationStatus.INDEX_FAILED))
-      }
-      return
+    if (pendingSafe?.safeAddress) {
+      pollSafeInfo(chainId, pendingSafe.safeAddress)
+        .then(() => setStatus(SafeCreationStatus.INDEXED))
+        .catch(() => setStatus(SafeCreationStatus.INDEX_FAILED))
```

Comment:
We should wait until `SUCCESS` before we start polling for the safe info otherwise it starts to poll even before the user has submitted the transaction in their wallet.